proposals/0130-dynamic-block-limits.md

```diff
@@ -0,0 +1,139 @@
+---
+simd: '0130'
+title: Dynamic Block Limits
+authors:
+  - Cavey Cool, Magnetar Fields
+category: Standard
+type: Core
+status: Draft
+created: 2024-03-24
+feature: (fill in with feature tracking issues once accepted)
+supersedes: (optional - fill this in if the SIMD supersedes a previous SIMD)
+extends: (optional - fill this in if the SIMD extends the design of a previous
+ SIMD)
+---
+
+## Summary
+
+This proposal introduces dynamic adjustments to the compute unit (CU) limit of
+Solana blocks based on network utilization at the end of each epoch. If the
+average block utilization exceeds 75%, the CU limit will increase by 20%;
+if it falls below 25%, the limit will decrease by 20%. A second metric based on
+vote slot latency is used to preserve protocol liveness and responsive UX. This
+proposal aims to optimize network performance by adapting to demonstrated
+compute capacity and demand without centralized decisions about limits and
+without voting.
+
+## Motivation
+
+The goal of this proposal is to enhance the scalability and efficiency of the
+Solana network by dynamically adjusting block CU limits in response to actual
+network demand, ensuring Solana scales with Moore's Law and exponentially
+increasing bandwidth.
+
+## Alternatives Considered
+
+**Manual Adjustment**: Periodic manual reviews of network performance to adjust
+CU limits. This lacks responsiveness, flexibility, and may be an arbitrary or
+centralized decision.
+
+## New Terminology
+
+- **Dynamic Block Limit**: Adjusting block CU limits based on network demand.
+- **Epoch Average Utilization**: The average CU utilization over an epoch.
+- **CU Limit Adjustment**: The process of modifying the block's CU limit.
+- **Supermajority Vote Slot Latency**: 67th percentile for the vote slot latency
+in an epoch.
+
+## Detailed Design
+
+### Implementation
+
+Record each block's CU utilization in an epoch and calculate the epoch average
+utilization (EAU). Record the slot latency of votes in an epoch and calculate the
+supermajority vote slot latency (SVSL).
+
+Increase CU limit by 20% if:
+
+1. EAU is greater than 75%, **AND**
+2. SVSL is below threshold vote slot latency (TBD).
+
+Decrease CU limit by 20% if:
+
+1. EAU is less than 25%, **OR**
```

Comment:
imo get rid of this. Network should redline what is possible based on today's available hardware and what validators are running. SVSL metric or some other proxy for cluster performance should be the only trigger for decreasing the CU limit